pandas/core/arrays/boolean.py

```diff
@@ -622,52 +620,6 @@ def _logical_method(self, other, op):
         # expected "ndarray"
         return BooleanArray(result, mask)  # type: ignore[arg-type]
 
-    def _cmp_method(self, other, op):
-        from pandas.arrays import (
-            FloatingArray,
-            IntegerArray,
-        )
-
-        if isinstance(other, (IntegerArray, FloatingArray)):
-            return NotImplemented
```

Comment:
Comparing the implementations, I think this is the only thing that's actually different / would change for BooleanArray. But since the implementations are otherwise equal, I suppose that doesn't matter?

Reply:
Yeah that was the same thought I had when looking at these